src/adapters/net.cpp

```diff
@@ -248,19 +253,21 @@ namespace net {
       return false;
     }
 
-    struct ifreq request {};
-    struct ethtool_cmd data {};
+    if(!m_bridge) { // If bridge network then link speed cannot be computed TODO: Identify the physical network in bridge and compute the link speed
+	  struct ifreq request {};
+	  struct ethtool_cmd data {};
 
-    memset(&request, 0, sizeof(request));
-    strncpy(request.ifr_name, m_interface.c_str(), IFNAMSIZ - 1);
-    data.cmd = ETHTOOL_GSET;
-    request.ifr_data = reinterpret_cast<char*>(&data);
+	  memset(&request, 0, sizeof(request));
+	  strncpy(request.ifr_name, m_interface.c_str(), IFNAMSIZ - 1);
+	  data.cmd = ETHTOOL_GSET;
+	  request.ifr_data = reinterpret_cast<char*>(&data);
 
-    if (ioctl(*m_socketfd, SIOCETHTOOL, &request) == -1) {
-      return false;
-    }
+	  if (ioctl(*m_socketfd, SIOCETHTOOL, &request) == -1) {
+	    return false;
+	  }
 
-    m_linkspeed = data.speed;
+	  m_linkspeed = data.speed;
```

Comment:
Instead of wrapping the whole thing in an if block, please only set the linkspeed if we don't use a bridge interface. We still need to check, if the interface is actually connected.

Also, please use 2 spaces for indentation, no tabs.

Reply:
I have made the changes as per your review comments